platform/android/src/native_map_view.cpp

```diff
@@ -827,47 +827,49 @@ void NativeMapView::addLayerAt(JNIEnv& env, jlong nativeLayerPtr, jni::jint inde
 /**
  * Remove by layer id.
  */
-jni::Local<jni::Object<Layer>> NativeMapView::removeLayerById(JNIEnv& env, const jni::String& id) {
+jni::jboolean NativeMapView::removeLayerById(JNIEnv& env, const jni::String& id) {
     std::unique_ptr<mbgl::style::Layer> coreLayer = map->getStyle().removeLayer(jni::Make<std::string>(env, id));
     if (coreLayer) {
-        return LayerManagerAndroid::get()->createJavaLayerPeer(env, *map, std::move(coreLayer));
-    } else {
-        return jni::Local<jni::Object<Layer>>();
+        LayerManagerAndroid::get()->createJavaLayerPeer(env, *map, std::move(coreLayer));
+        return jni::jni_true;
     }
+    return jni::jni_false;
 }
 
 /**
  * Remove layer at index.
  */
-jni::Local<jni::Object<Layer>> NativeMapView::removeLayerAt(JNIEnv& env, jni::jint index) {
+jni::jboolean NativeMapView::removeLayerAt(JNIEnv& env, jni::jint index) {
     auto layers = map->getStyle().getLayers();
 
     // Check index
     int numLayers = layers.size() - 1;
     if (index > numLayers || index < 0) {
         Log::Warning(Event::JNI, "Index out of range: %i", index);
-        return jni::Local<jni::Object<Layer>>();
+        return jni::jni_false;
     }
 
     std::unique_ptr<mbgl::style::Layer> coreLayer = map->getStyle().removeLayer(layers.at(index)->getID());
     if (coreLayer) {
-        return LayerManagerAndroid::get()->createJavaLayerPeer(env, *map, std::move(coreLayer));
-    } else {
-        return jni::Local<jni::Object<Layer>>();
+        LayerManagerAndroid::get()->createJavaLayerPeer(env, *map, std::move(coreLayer));
+        return jni::jni_true;
     }
+    return jni::jni_false;
 }
 
 /**
  * Remove with wrapper object id. Ownership is transferred back to the wrapper
  */
-void NativeMapView::removeLayer(JNIEnv&, jlong layerPtr) {
+jni::jboolean NativeMapView::removeLayer(JNIEnv&, jlong layerPtr) {
```

Comment:
Every other `remove` method besides this one drops the reference to the layer, so the one kept in Java is probably going to become invalid. In order to make our intended logic possible, that a user fetches the layer with `getLayer` (or just keeps the reference from the start) and can keep after it's removed from the map (successfully or not), we need to keep the reference valid.